src/consensus/pbft/libbyz/message.cpp

```diff
@@ -50,50 +50,43 @@ Message::Message(int t, unsigned sz)
 Message::Message(Message_rep* cont)
 {
   PBFT_ASSERT(ALIGNED(cont), "Improperly aligned pointer");
-  msg = cont;
-  max_size = -1; // To prevent contents from being deallocated or trimmed
+  int max_size = -1; // To prevent contents from being deallocated or trimmed
+  msg = std::move(make_Ref<MsgBufCounter>(cont, max_size, false));
+  msg_buf = cont;
   auth_type = Auth_type::unknown;
   auth_len = 0;
   auth_dst_offset = 0;
   next = nullptr;
-  should_delete = false;
-}
-
-Message::~Message()
-{
-  if (max_size > 0 && msg != nullptr)
-  {
-    free(msg);
-  }
 }
 
 void Message::trim()
 {
-  if (max_size > 0)
+  if (msg->max_size > 0)
   {
-    max_size = msg->size;
+    msg->max_size = msg_buf->size;
   }
 }
 
 void Message::set_size(int size)
 {
-  PBFT_ASSERT(msg && ALIGNED(msg), "Invalid state");
-  if (!(max_size < 0 || ALIGNED_SIZE(size) <= max_size))
+  PBFT_ASSERT(msg_buf && ALIGNED(msg_buf), "Invalid state");
+  if (!(msg->max_size < 0 || ALIGNED_SIZE(size) <= msg->max_size))
   {
     LOG_INFO << "Error - size:" << size
              << ", aligned_size:" << ALIGNED_SIZE(size)
-             << ", max_size:" << max_size << std::endl;
+             << ", max_size:" << msg->max_size << std::endl;
   }
-  PBFT_ASSERT(max_size < 0 || ALIGNED_SIZE(size) <= max_size, "Invalid state");
+  PBFT_ASSERT(
+    msg->max_size < 0 || ALIGNED_SIZE(size) <= msg->max_size, "Invalid state");
   int aligned = ALIGNED_SIZE(size);
   for (int i = size; i < aligned; i++)
   {
-    ((char*)msg)[i] = 0;
+    ((char*)msg_buf)[i] = 0;
   }
-  msg->size = aligned;
+  msg_buf->size = aligned;
 }
 
-bool Message::convert(char* src, unsigned len, int t, int sz, Message& m)
+bool Message::convert(char* src, unsigned len, int t, int sz, Message& msg_buf)
```

Comment:
Minor - I think this should still be `m`. It looks like it shadows the member variable `msg_buf`, it doesn't actually because this is static, but that's non-obvious. But everywhere else, `msg_buf` refers to the `Message_rep*` member variable, so this `Message&` argument should be given a distinct name.